Documentation/ApiOverview/ContentElements/CustomDataProcessing.rst

```diff
@@ -1,160 +1,159 @@
+    Interface; DataProcessorInterface
+..  _content-elements-custom-data-processor_implementation:
 
 Implementing the custom data processor
 ======================================
 
-The custom data processor needs to implement :php:`DataProcessorInterface`.
-The main function :php:`process` gets called with the following parameters:
+The custom data processor must implement :php:`\TYPO3\CMS\Frontend\ContentObject\DataProcessorInterface`.
+The main method :php:`process()` gets called with the following parameters:
 
 :php:`ContentObjectRenderer $cObj`
-   Receives the data of the current TypoScript context, in this case the
-   data of the calling content element.
+    Receives the data of the current TypoScript context, in this case the
+    data of the calling content element.
 
 :php:`array $contentObjectConfiguration`
-   Contains the configuration of the calling content element. In this example
-   all configuration of :typoscript:`tt_content.examples_dataproccustom`
+    Contains the configuration of the calling content element. In this example
+    all configuration of :typoscript:`tt_content.examples_dataproccustom`
 
 :php:`array $processorConfiguration`
-   Contains the configuration of the currently called dataprocessor. In this
-   case the value of :typoscript:`as` and the stdWrap configuration of the
-   :typoscript:`categoryList`
+    Contains the configuration of the currently called data processor. In this
+    case the value of :typoscript:`as` and the :typoscript:`stdWrap`
+    configuration of the :typoscript:`categoryList`
 
 
 :php:`array $processedData`
-   On calling, contains the processed data of all previously called data
-   processors on this content element. Your custom data processor also stores
-   the variables to be send to Fluid here.
+    On calling, contains the processed data of all previously called data
+    processors on this content element. Your custom data processor also stores
+    the variables to be send to Fluid here.
 
 This is an example implementation of a custom data processor:
 
-.. code-block:: php
-   :caption: EXT:examples/Classes/DataProcessing/CustomCategoryProcessor.php
-
-   namespace T3docs\Examples\DataProcessing;
-
-   use TYPO3\CMS\Core\Utility\GeneralUtility;
-   use TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer;
-   use TYPO3\CMS\Frontend\ContentObject\DataProcessorInterface;
-   use TYPO3\CMS\Extbase\Domain\Repository\CategoryRepository;
-
-   class CustomCategoryProcessor implements DataProcessorInterface
-   {
-      public function process(
-         ContentObjectRenderer $cObj,
-         array $contentObjectConfiguration,
-         array $processorConfiguration,
-         array $processedData
-      ) : array {
-         if (isset($processorConfiguration['if.']) && !$cObj->checkIf($processorConfiguration['if.'])) {
-            // leave $processedData unchanged in case there were previous other processors
+..  code-block:: php
```

Comment:
This code is found in the examples extension, can we generate the snippet?

Reply:
In the next step for processor aliases in main (as they already have the aliases in place). This is just a preparation for that step to have small PRs.